node/node.go

```diff
@@ -0,0 +1,120 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/log"
+	ethNode "github.com/ethereum/go-ethereum/node"
+	"github.com/status-im/status-go/geth/params"
+	"github.com/status-im/status-go/geth/signal"
+)
+
+// node-related errors
+var (
+	ErrEthServiceRegistrationFailure     = errors.New("failed to register the Ethereum service")
+	ErrWhisperServiceRegistrationFailure = errors.New("failed to register the Whisper service")
+	ErrLightEthRegistrationFailure       = errors.New("failed to register the LES service")
+	ErrNodeMakeFailure                   = errors.New("error creating p2p node")
+	ErrNodeRunFailure                    = errors.New("error running p2p node")
+	ErrNodeStartFailure                  = errors.New("error starting p2p node")
+)
+
+type Node interface {
+	Config() *params.NodeConfig
+	Node() *ethNode.Node
+	Start() (chan struct{}, error)
```

Comment:
In this refactoring, we'd like to get rid of starting a node asynchronously. The API exposed by Node should be the simplest possible and if the consumer would like to start it asynchronously, it should handle that instead of implementing it in the Node.